element-connector/src/main/java/org/eclipse/californium/elements/PrincipalEndpointContextMatcher.java

```diff
@@ -24,14 +24,34 @@
  */
 public class PrincipalEndpointContextMatcher implements EndpointContextMatcher {
 
+	private final boolean principalAsPeer;
+
 	public PrincipalEndpointContextMatcher() {
+		this(false);
+	}
+
+	public PrincipalEndpointContextMatcher(boolean serverSide) {
+		principalAsPeer = serverSide;
```

Comment:
It works on the server side.

Reply:
For me this is not really about "server side". and I feel that "server side" could be confusing in coap because often peers play both role.

the boolean could be renamed by `usePrincipalAsIdentity` or something like this.

Reply:
(ping just in case you missed this comment)

Reply:
I rename it.